### PR TITLE
Show direction glyph toward flag halfway out, save and Mac delete enhancements

### PR DESCRIPTION
**Pull request by frin**
This is a similar behavior as seen in original S2. (Drawing while in road building mode)

### CODE REVIEW

**Comment by Flow86** [comment]
can you please provide a screenshot? I cant imagine it ;-)

**Comment by frin** [comment]
Yes of course, marked with magenta arrows http://i.imgur.com/QdQWpUe.png

Previously you wouldn't know the slope of the road for the last road segment.

**Comment by frin** [comment]
Oh I guess this gets added automatically to existing PR. Well, there's more enhancements. I updated the PR title

**Comment by Spikeone** [comment]
Do you have a screenshot with extreme heights? Showing what that looks like? Your current screenshot shows quite moderate height differences

**Comment by frin** [comment]
Yeah here is the biggest height difference, it is slightly overlapping, but so does the original S2, screenshot for comparison. Still it's not that bad.
s2_height

File: src/GameClient.cpp [hunk 1]
```diff
@@ -1684,9 +1684,16 @@ unsigned GameClient::SaveToFile(const std::string& filename)
 
     // Und alles speichern
     if(!save.Save(filename))
+    {
+        GameMessage_System_Chat saveAnnouncement = GameMessage_System_Chat(playerId_, "Error occurred while saving game.");
```
Comment (Flamefire) [inline src/GameClient.cpp]: Not a good idea: Saving is done per client. You would most likely flood the game with messages from every player. Rather just show it to the local player maybe using the notification system.
Also use the localization feature: `_("Translated text")`

File: src/ingameWindows/iwSave.cpp [hunk 2]
```diff
@@ -127,7 +127,13 @@ void iwSave::SaveLoad()
     tmp += ".sav";
 
     // Speichern
-    GAMECLIENT.SaveToFile(tmp);
+    int error = GAMECLIENT.SaveToFile(tmp);
+    if (error == 0)
```
Comment (Flamefire) [inline src/ingameWindows/iwSave.cpp]: Maybe this would be the place to show the message?

File: src/world/GameWorldView.cpp [hunk 3]
```diff
@@ -308,7 +308,21 @@ void GameWorldView::DrawGUI(const RoadBuildState& rb, const TerrainRenderer& ter
...
                     }
 
-                    LOADER.GetMapImageN(id)->DrawFull(curPos);
+                    // If item at direction is a flag and not the start, draw the direction glyph halfway out
+                    if (GetWorld().GetNO(curPt)->GetType() == NOP_FLAG && curPt != rb.start)
+                    {
+                        // Last clicked position (coordinate) in the road route
+                        Point<int> lastPos = GetWorld().GetNodePos(rb.point) - offset + curOffset;
+                        Point<int> halfwayPos;
+                        halfwayPos.x = (curPos.x + lastPos.x) / 2;
```
Comment (Flamefire) [inline src/world/GameWorldView.cpp]: Easier: Replace `Point<int>` by `DrawPoint` and use `halfwayPos = (curPos + lastPos) / 2`

File: src/world/GameWorldView.cpp [hunk 4]
```diff
@@ -290,7 +290,7 @@ void GameWorldView::DrawGUI(const RoadBuildState& rb, const TerrainRenderer& ter
                     continue;
 
                 if((gwv.IsRoadAvailable(rb.mode == RM_BOAT, curPt) && gwv.IsOwner(curPt) && GetWorld().IsPlayerTerritory(curPt))
-                   || (gwv.GetBQ(curPt) == BQ_FLAG))
+                   || (gwv.GetBQ(curPt) == BQ_FLAG) || (GetWorld().GetNO(curPt)->GetType() == NOP_FLAG && curPt != rb.start))
```
Comment (Flamefire) [inline src/world/GameWorldView.cpp]: Can the last condition become a problem if the flag belongs to an other player? Maybe the last 2 conditions (Owner and player territory) should always apply. But would need to think carefully about this.

**Comment by frin** [comment]
Interesting suggestions (didn't know saves are separate for each client). I will look into these and update code accordingly.

**Comment by Flamefire** [comment]
@frin Are you still working on that? Would love to see this changes merged :)

**Comment by Shawn8901** [comment]
@Flamefire  I think this PR can be closed, as every change was merged via #821

**Pull request closed without merging**